### ISSUE TITLE
[stock_valuation_account_manual_adjustment] Wrong Inventory Value with Standard Price

### ISSUE THREAD

**Issue by travish2** [OCA/stock-logistics-warehouse#355]
Apologies if I have missed something 

The issue:
The Inventory Value in stock_valuation_account_manual_adjustment of any given product is doubled when using Standard Costing but correct if it is switched to Average Costing. This is on a v9 database

Eg: Product A has a cost of $1000 and there is 3 on hand. Average costing shows Inventory Value at $3000, Standard Costing shows $6000

I believe the problem is in this section of code from models/product.py:

```
            Q5 AS (
                SELECT pr.id, sum(sq.qty*ip.value_float) as valuation
                FROM stock_quant as sq
                INNER JOIN product_product as pr
                ON pr.id = sq.product_id
                INNER JOIN stock_location as sl
                ON sl.id = sq.location_id
                INNER JOIN Q3
                ON (Q3.id = pr.product_tmpl_id
                AND Q3.cost_method in ('standard', 'average'))
                INNER JOIN ir_property as ip
                ON (ip.res_id = 'product.product,' || pr.id)
                INNER JOIN ir_model_fields imf
                ON (imf.id = ip.fields_id
                AND imf.name = 'standard_price')
                WHERE sl.usage = 'internal'
                GROUP BY pr.id),
```

My SQL is a little rusty so this is where I am stuck but I believe the issue is in the last few lines.

If the first line is modified to return just ip.value_float it returns the correct cost when using average ($1000) but the doubled cost with standard ($2000)

**Comment by travish2** [comment]
An update:

It looks like the issue is actually from the Q2 portion of code - when stepping through it manually it came out with 2 rows rather than 1, which then doubled the result later on.
With average costing Q1 returns a single row so that stays correct.

The double rows in Q2 were caused by ir_model_fields matching product_template.id and "property_cost_method" on two records.

As an attempted fix I have added DISTINCT statements to Q3:
```            Q3 AS (
                SELECT DISTINCT * FROM Q1
                UNION ALL
                SELECT DISTINCT * FROM Q2),
```

Open to any feedback.
